### PR TITLE
dbt library doc update

### PR DESCRIPTION
**Pull request by makotonium**
## Summary
This is a follow-up documentation changes for [PR 4202](https://github.com/dagster-io/dagster/pull/4202) for `dagster-dbt` library, and includes other minor edits to other parts of the documentation for missing info that would have been nice to have or if additional clarification would have been useful.

## Test Plan
NA

## Checklist

- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [NA] I have added tests to cover my changes.

### CODE REVIEW

**Comment by makotonium** [comment]
@OwenKephart This PR contains the documentation changes that I wanted to include for the `dagster-dbt` PR that I did recently.

File: docs/content/integrations/dbt.mdx [hunk 1]
```diff
@@ -58,6 +58,26 @@ def my_dbt_pipeline():
 
 In the code snippet above, the config `"models"` takes a list of strings. The string `"tag:staging"` uses [dbt's node selection syntax](https://docs.getdbt.com/reference/node-selection/syntax) to filter models with the tag `"staging"`. For more details, [visit the official dbt documentation on the node selection syntax](https://docs.getdbt.com/reference/node-selection/syntax).
 
+**Example:** The solid `dbt_cli_run` is configured to run specific models similar to above except after another solid has executed. When use the `start_after` argument, which has the `InputDefinitions` of `dagster_type=Nothing`, you can assign the output from another solid so that you can still model the dependencies with another solid. See the [Order Based Dependencies](/concepts/solids-pipelines/pipelines#order-based-dependencies-nothing-dependencies) for more info.
+```python file=/integrations/dbt.py startafter=start_marker_dbt_cli_run_after_another_solid endbefore=end_marker_dbt_cli_run_after_another_solid dedent=4
```
Comment (OwenKephart) [inline docs/content/integrations/dbt.mdx:62]: I don't see the corresponding change for this snippet in dagster/examples/docs_snippets/docs_snippets/integrations/dbt.py. These snippets are autogenerated (using the yarn snapshot command), and you can look in that file to see some examples (or maybe you just forgot to add those changes to this diff).

File: docs/content/integrations/dbt.mdx [hunk 2]
```diff
@@ -58,6 +58,26 @@ def my_dbt_pipeline():
 
 In the code snippet above, the config `"models"` takes a list of strings. The string `"tag:staging"` uses [dbt's node selection syntax](https://docs.getdbt.com/reference/node-selection/syntax) to filter models with the tag `"staging"`. For more details, [visit the official dbt documentation on the node selection syntax](https://docs.getdbt.com/reference/node-selection/syntax).
 
+**Example:** The solid `dbt_cli_run` is configured to run specific models similar to above except after another solid has executed. When use the `start_after` argument, which has the `InputDefinitions` of `dagster_type=Nothing`, you can assign the output from another solid so that you can still model the dependencies with another solid. See the [Order Based Dependencies](/concepts/solids-pipelines/pipelines#order-based-dependencies-nothing-dependencies) for more info.
```
Comment (OwenKephart) [inline docs/content/integrations/dbt.mdx:61]: I'd suggest some slight changes to the wording just for clarity, maybe something like:

Similar to the above example, the solid `dbt_cli_run` is specified to run after another solid has executed. In your pipeline definition, you can assign the output from an upstream solid to the `start_after` argument of your dbt solid, which will model that dependency relationship for you. See the documentation for [Order Based Dependencies](/concepts/solids-pipelines/pipelines#order-based-dependencies-nothing-dependencies) for more info.

**Review by OwenKephart — CHANGES_REQUESTED** [review]
Really appreciate you taking the time to add in the documentation! Had some minor comments but overall looks great 😄

File: docs/content/integrations/dbt.mdx (hunk 1 above)
Comment (makotonium) [reply docs/content/integrations/dbt.mdx:62]: Nope. Just had no idea what I was supposed to do. Will get this fixed.

**Comment by makotonium** [comment]
@OwenKephart This is ready for re-review. I've addressed the changes requested.

**Review by OwenKephart — CHANGES_REQUESTED** [review]
Awesome! Thanks again for contributing :)

This should be explained in the contributing guide, but do you mind running a quick

`make black` in the top level dagster directory and then

`yarn format` in dagster/docs/next?

Just some tiny formatting issues that our tests are complaining about (it's a pain, I know). Sorry for the trouble! After that, I'll go ahead and merge these in for you.

**Comment by makotonium** [comment]
Ooops thanks for the reminder. No problem. I ran both and it looks like all the checks passed. I'll configure my IDE so I don't forget next time!

**Comment by OwenKephart** [comment]
awesome! merging this in now, thanks again 😃

**Pull request merged**